libs/utils/src/ostream.cpp

```diff
@@ -35,6 +38,31 @@ namespace utils::io {
 
 ostream::~ostream() = default;
 
+void ostream::setConsumer(ConsumerCallback consumer, void* user) noexcept {
+    auto* const pImpl = mImpl;
+    std::lock_guard const lock(pImpl->mLock);
+    pImpl->mConsumer = { consumer, user };
+}
+
+ostream& flush(ostream& s) noexcept {
+    auto* const pImpl = s.mImpl;
+    pImpl->mLock.lock();
+    auto const callback = pImpl->mConsumer;
+    if (UTILS_UNLIKELY(callback.first)) {
+        auto& buf = s.getBuffer();
+        char* const str = strdup(buf.get());
```

Comment:
Can buf.get() be null?

Reply:
ah good remark. I think it can technically be `null` if the log has never been written to, should be rare, but still.

Reply:
okay, fixed a couple nullptr issues.